packages/core/src/generator/generator.service.ts

```diff
@@ -11,6 +11,38 @@ export class GeneratorService {
     @multiInject(GENERATOR_ADAPTER_IDENTIFIER) private readonly generatorAdapters: GeneratorAdapter[]
   ) { }
 
+  /**
+   * Get list of supported CI/CD platforms based on registered generator adapters
+   */
+  getSupportedCicdPlatforms(): string[] {
+    return this.generatorAdapters.map(adapter => adapter.getPlatformName());
+  }
+
+  /**
+   * Get generator adapter for a specific platform
+   */
+  getGeneratorAdapterByPlatform(platform: string): GeneratorAdapter | undefined {
+    return this.generatorAdapters.find(adapter => adapter.getPlatformName() === platform);
+  }
+
+  /**
+   * Get supported sections for a specific CI/CD platform
+   */
+  getSupportedSectionsForPlatform(platform: string): string[] {
+    const adapter = this.getGeneratorAdapterByPlatform(platform);
+    return adapter ? adapter.getSupportedSections() : [];
+  }
+
+  /**
+   * Get all supported sections from all registered generator adapters
+   */
+  getAllSupportedSections(): string[] {
+    const allSections = new Set<string>();
+    for (const adapter of this.generatorAdapters) {
+      adapter.getSupportedSections().forEach(section => allSections.add(section));
+    }
+    return Array.from(allSections).sort();
+  }
```

Comment:
It is useless because we have always a cicd platform defined (settled by option or guessed by the source option)